modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/continuous/GridCacheContinuousQueryAbstractSelfTest.java

```diff
@@ -207,6 +207,11 @@ protected NearCacheConfiguration nearConfiguration() {
 
                     if (!((Map)U.field(proc, "rmtInfos")).isEmpty())
                         return false;
+
+                    CacheContinuousQueryManager mgr = grid(i).context().cache().internalCache(DEFAULT_CACHE_NAME).context().continuousQueries();
+
+                    if (!((Map)U.field(mgr, "lsnrs")).isEmpty())
```

Comment:
I see the same check a bit lines below, why do we need it here ?

Reply:
Without this check, the race happens because of the concurrency nature. We reach the second `lsnrs` assert expecting it's empty (it should) but as we add logging sometimes it take a liitle bit much time to empty listeners fails. Its a race condition.

Reply:
If i correctly understand your answer - this fix is not corresponding to initial issue about logging, just fix flaky test, is it correct ?

Reply:
Yes, it's correct, just a fix of flaky test